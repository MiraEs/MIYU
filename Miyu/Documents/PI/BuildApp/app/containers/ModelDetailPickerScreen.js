'use strict';

import React, {
	Component,
} from 'react';
import PropTypes from 'prop-types';
import {
	StyleSheet,
	View,
} from 'react-native';
import helpers from '../lib/helpers';
import NavigationBar from '../components/NavigationBar';
import Pager from '../components/Library/Pager/Pager';
import styles from '../lib/styles';
import {
	Button,
	Text,
} from 'BuildLibrary';
import { getSelectedModel } from '../actions/UpsellActions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import trackingActions from '../lib/analytics/TrackingActions';
import { getProductSpecs } from '../actions/ProductDetailActions';
import ModelDetailPage from '../components/ModelDetailPage';
import { trackState } from '../actions/AnalyticsActions';

const {
	width,
} = styles.dimensions;

const componentStyles = StyleSheet.create({
	mainWrapper: {
		flex: 1,
		backgroundColor: styles.colors.white,
	},
	header: {
		flexDirection: 'row',
		paddingHorizontal: styles.measurements.gridSpace1,
		paddingVertical: styles.measurements.gridSpace2,
		backgroundColor: styles.colors.lightGray,
	},
	pagerStyles: {
		width,
		flex: 1,
	},
	markerStyle: {
		borderTopWidth: styles.dimensions.borderWidth,
		borderTopColor: styles.colors.lightGray,
	},
});

export class ModelDetailPickerScreen extends Component {
	constructor(props) {
		super(props);
		this._currentPageIndex = 0;
	}

	componentDidMount() {
		this.props.actions.trackState('build:app:modeldetailpicker');
		this.onPageChanged(0);
	}

	onBackPress = () => {
		this.props.navigator.pop();
	};

	onPageChanged = (curIndex) => {
		this._currentPageIndex = curIndex;
		const { actions, optionProducts, productsSpecs } = this.props;
		const currentProduct = optionProducts[curIndex];
		const compositeId = currentProduct.productDrop.productCompositeId;
		if (!productsSpecs || !productsSpecs[compositeId]) {
			actions.getProductSpecs(compositeId).done();
		}
	};

	choose = () => {
		const { actions, finish, optionId } = this.props;
		const selectedProduct = this.props.optionProducts[this._currentPageIndex];
		actions.getSelectedModel({
			finish,
			optionId,
			selectedProductCompositeId: selectedProduct.productDrop.productCompositeId,
		});
		this.props.navigator.pop();
	};

	renderPage = () => {
		const { optionProducts, productsSpecs } = this.props;
		return optionProducts.map((product, index) => {
			const specs = productsSpecs ? productsSpecs[product.productDrop.productCompositeId] : [];
			return (
				<ModelDetailPage
					key={index}
					finish={this.props.finish}
					product={product}
					productSpecs={specs}
				/>
			);
		});
	};

	render() {
		return (
			<View style={componentStyles.mainWrapper}>
				<NavigationBar
					leftNavButton={{
						icon: helpers.getIcon('arrow-back'),
						onPress: this.onBackPress,
					}}
				/>

				<View style={componentStyles.header}>
					<Text>Select </Text>
					<Text weight="bold">{this.props.productFriendlyName}</Text>
				</View>
				<Pager
					onPageChanged={this.onPageChanged}
					style={componentStyles.pagerStyles}
					markerStyle={componentStyles.markerStyle}
				>
					{this.renderPage()}
				</Pager>
				<View>
					<Button
						text="Select"
						onPress={this.choose}
						trackAction={trackingActions.RELATED_PRODUCT_CHOOSE_MODEL}
						accessibilityLabel="Choose Button"
					/>
				</View>
			</View>
		);
	}
}

ModelDetailPickerScreen.propTypes = {
	actions: PropTypes.object,
	finish: PropTypes.string,
	optionId: PropTypes.string.isRequired,
	optionProducts: PropTypes.array.isRequired,
	productFriendlyName: PropTypes.string.isRequired,
	productsSpecs: PropTypes.object,
	navigator: PropTypes.shape({
		pop: PropTypes.func,
	}),
};

const mapStateToProps = (state) => {
	return {
		productsSpecs: state.productDetailReducer.productsSpecs,
	};
};

const mapDispatchToProps = (dispatch) => {
	return {
		actions: bindActionCreators({
			getSelectedModel,
			getProductSpecs,
			trackState,
		}, dispatch),
	};
};

export default connect(mapStateToProps, mapDispatchToProps)(ModelDetailPickerScreen);
