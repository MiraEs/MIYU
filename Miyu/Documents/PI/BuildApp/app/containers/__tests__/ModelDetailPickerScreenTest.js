'use strict';

jest.mock('../../../app/services/httpClient', () => ({}));
jest.mock('../../../app/store/configStore', () => ({}));
jest.mock('../../../app/lib/analytics/tracking');
jest.mock('BuildNative');
jest.mock('BuildLibrary');
jest.mock('../../../app/components/Form', () => 'Form');
jest.mock('../../../app/components/FormInput', () => 'FormInput');

jest.unmock('react-native');

import { ModelDetailPickerScreen } from '../ModelDetailPickerScreen';
import React from 'react';

const defaultProps = {
	actions: {
		trackState: jest.fn(),
		getProductSpecs: jest.fn(() => ({ done: jest.fn() })),
	},
	optionProducts: [
		{
			productDrop: {
				finishes: [{}],
			},
		},
	],
};

describe('ModelDetailPickerScreen component', () => {

	it('should render correctly', () => {
		const tree = require('react-test-renderer').create(
			<ModelDetailPickerScreen {...defaultProps} />
		).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
