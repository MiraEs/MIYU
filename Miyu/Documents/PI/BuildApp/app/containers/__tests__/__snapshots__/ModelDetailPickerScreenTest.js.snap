// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelDetailPickerScreen component should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#63666A",
          "height": 64,
          "shadowColor": "black",
        },
        Object {},
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
          },
          Object {
            "marginTop": 20,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel="Left Nav Button"
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={
            Object {
              "bottom": 14,
              "left": 14,
              "right": 14,
              "top": 14,
            }
          }
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "minHeight": 44,
              "minWidth": 44,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Icon
            color="#FFFFFF"
            name="ios-arrow-back"
            style={
              Array [
                Object {
                  "fontSize": 28,
                },
                Object {},
              ]
            }
          />
          <Text
            color="white"
            lineHeight={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "ProximaNova-Regular",
                  "fontSize": 16,
                  "marginVertical": 8,
                  "textAlign": "center",
                },
                Object {
                  "paddingLeft": 8,
                },
                Object {},
                undefined,
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
            "paddingRight": 8,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F2F2F2",
        "flexDirection": "row",
        "paddingHorizontal": 8,
        "paddingVertical": 16,
      }
    }
  >
    <Text>
      Select 
    </Text>
    <Text
      weight="bold"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "width": 320,
      }
    }
  >
    <ListView
      contentOffset={
        Object {
          "x": 0,
        }
      }
      dataSource={
        Object {
          "_data": Array [
            <View
              onLayout={[Function]}
        >
              <ModelDetailPage
                    finish={undefined}
                    product={
                          Object {
                                "productDrop": Object {
                                  "finishes": Array [
                                    Object {},
                                  ],
                                },
                              }
                    }
                    productSpecs={Array []}
              />
        </View>,
          ],
          "getRowCount": [Function],
        }
      }
      enableEmptySections={true}
      horizontal={true}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      pagingEnabled={true}
      renderRow={[Function]}
      scrollEnabled={undefined}
      showsHorizontalScrollIndicator={false}
      snapToInterval={1}
      style={
        Object {
          "flex": 1,
          "width": 320,
        }
      }
    >
      <View
        onLayout={[Function]}
      >
        <ScrollView
          enableEmptySections={true}
        >
          <View
            style={
              Object {
                "paddingHorizontal": 16,
                "width": 320,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <Image
                height={175}
                source="https://s3.img-b.com/image/upload/t_base,f_auto,dpr_2.0,c_lpad,w_175,h_175/404images/noimage.gif"
                width={175}
              />
            </View>
            <Text
              style={
                Object {
                  "marginTop": 16,
                }
              }
              weight="bold"
            >
               
               
               
            </Text>
            <Text
              color="primary"
              style={
                Object {
                  "marginTop": 8,
                }
              }
              weight="bold"
            />
          </View>
        </ScrollView>
      </View>
    </ListView>
  </View>
  <View>
    <Button
      accessibilityLabel="Choose Button"
      onPress={[Function]}
      text="Select"
      trackAction="build:app:relatedproduct:choosemodel"
    />
  </View>
</View>
`;
